The `freeimage` package does not build on M1 Macbook

### Describe the bug
The `freeimage` package does not build on M1 Macbook. The error message is 
```
this derivation will be built:
  /nix/store/pbrfippl3qwri32nimx5k0nv28zvhkfw-freeimage-unstable-2020-07-04.drv
building '/nix/store/pbrfippl3qwri32nimx5k0nv28zvhkfw-freeimage-unstable-2020-07-04.drv'...
unpacking sources
unpacking source archive /nix/store/dpsnbzmcbaqnk1nmhxzfll66vqf5jibs-svn-r1859
source root is svn-r1859/FreeImage/trunk
patching sources
applying patch /nix/store/rbvlicqsmhlcyd63mgyhc426i1vhlvb4-unbundle.diff
patching file Makefile.fip
patching file Makefile.gnu
patching file Makefile.osx
patching file Makefile.srcs
patching file Source/FreeImage.h
patching file Source/FreeImage/J2KHelper.cpp
patching file Source/FreeImage/PluginEXR.cpp
patching file Source/FreeImage/PluginG3.cpp
patching file Source/FreeImage/PluginJ2K.cpp
patching file Source/FreeImage/PluginJP2.cpp
patching file Source/FreeImage/PluginJPEG.cpp
patching file Source/FreeImage/PluginJXR.cpp
patching file Source/FreeImage/PluginPNG.cpp
patching file Source/FreeImage/PluginRAW.cpp
patching file Source/FreeImage/PluginTIFF.cpp
patching file Source/FreeImage/PluginWebP.cpp
patching file Source/FreeImage/ZLibInterface.cpp
patching file Source/FreeImageToolkit/JPEGTransform.cpp
patching file Source/Metadata/XTIFF.cpp
patching file fipMakefile.srcs
updateAutotoolsGnuConfigScriptsPhase
configuring
no configure script, doing nothing
building
build flags: -j8 -l8 SHELL=/nix/store/wbwcrf1g7ywmmw2lllc26r0zcdvassiq-bash-5.1-p8/bin/bash
make -f Makefile.osx
make[1]: Entering directory '/private/tmp/nix-build-freeimage-unstable-2020-07-04.drv-0/svn-r1859/FreeImage/trunk'
clang++ -Wno-ctor-dtor-privacy -D__ANSI__ -std=c++11 -stdlib=libc++ -Wc++11-narrowing -Os -fexceptions -fvisibility=hidden -DNO_LCMS -D__ANSI__ -arch x86_64 -I. -ISource -ISource/Metadata -ISource/FreeImageToolkit -D__ANSI__ -DDISABLE_PERF_MEASUREMENT -I/nix/store/cilmyphkp4qylccn2fg5yilfcm47vmb0-libtiff-4.2.0-dev/include -I/nix/store/b9kr3r3i8fawppc03n053lqf58lz72g2-libjpeg-turbo-2.1.0-dev/include -I/nix/store/wnqimqy7nq7rbpf76jw049ilp2s4xhh0-zlib-1.2.11-dev/include -I/nix/store/bz6y9r2rmmramlavyr5l1r4wsr67swkq-libpng-apng-1.6.37-dev/include/libpng16 -I/nix/store/aff961qxn8w5yn8vh5n1xd5isfsvbwhb-libwebp-1.1.0/include -I/nix/store/b96jp9m78pkkazxpqj1flbi87510w7x9-libraw-0.20.2-dev/include/libraw -I/nix/store/b96jp9m78pkkazxpqj1flbi87510w7x9-libraw-0.20.2-dev/include -I/nix/store/4j2ydnq9fkic9m8x0cqwqlman6qf8f3c-lcms2-2.12-dev/include -I/nix/store/6w8vp1adlq4yqfvfpi44fgla2ms68bsm-openexr-2.5.7-dev/include/OpenEXR -I/nix/store/6w8vp1adlq4yqfvfpi44fgla2ms68bsm-openexr-2.5.7-dev/include -I/nix/store/6w8vp1adlq4yqfvfpi44fgla2ms68bsm-openexr-2.5.7-dev/include/OpenEXR -I/nix/store/3b9cki7cd7i90z7sjmcsrjpmw1v1nl9c-openjpeg-2.4.0-dev/include/openjpeg-2.4 -I/nix/store/6f8f0ny6z1cgw7pxr2a398fjl87y2jk2-jxrlib-1.1/include/libjxr/common -I/nix/store/6f8f0ny6z1cgw7pxr2a398fjl87y2jk2-jxrlib-1.1/include/libjxr/image/x86 -I/nix/store/6f8f0ny6z1cgw7pxr2a398fjl87y2jk2-jxrlib-1.1/include/libjxr/image -I/nix/store/6f8f0ny6z1cgw7pxr2a398fjl87y2jk2-jxrlib-1.1/include/libjxr/glue -I/nix/store/6f8f0ny6z1cgw7pxr2a398fjl87y2jk2-jxrlib-1.1/include/libjxr/test  -mmacosx-version-min=10.11 -c Source/FreeImage/BitmapAccess.cpp -o Source/FreeImage/BitmapAccess.o-x86_64
error: unknown target CPU 'armv8.3-a+crypto+sha2+aes+crc+fp16+lse+simd+ras+rdm+rcpc'
note: valid target CPU values are: nocona, core2, penryn, bonnell, atom, silvermont, slm, goldmont, goldmont-plus, tremont, nehalem, corei7, westmere, sandybridge, corei7-avx, ivybridge, core-avx-i, haswell, core-avx2, broadwell, skylake, skylake-avx512, skx, cascadelake, cooperlake, cannonlake, icelake-client, icelake-server, tigerlake, knl, knm, k8, athlon64, athlon-fx, opteron, k8-sse3, athlon64-sse3, opteron-sse3, amdfam10, barcelona, btver1, btver2, bdver1, bdver2, bdver3, bdver4, znver1, znver2, x86-64
make[1]: *** [Makefile.osx:104: Source/FreeImage/BitmapAccess.o-x86_64] Error 1
make[1]: Leaving directory '/private/tmp/nix-build-freeimage-unstable-2020-07-04.drv-0/svn-r1859/FreeImage/trunk'
make: *** [Makefile:21: default] Error 2
error: builder for '/nix/store/pbrfippl3qwri32nimx5k0nv28zvhkfw-freeimage-unstable-2020-07-04.drv' failed with exit code 2;
       last 10 log lines:
       > building
       > build flags: -j8 -l8 SHELL=/nix/store/wbwcrf1g7ywmmw2lllc26r0zcdvassiq-bash-5.1-p8/bin/bash
       > make -f Makefile.osx
       > make[1]: Entering directory '/private/tmp/nix-build-freeimage-unstable-2020-07-04.drv-0/svn-r1859/FreeImage/trunk'
       > clang++ -Wno-ctor-dtor-privacy -D__ANSI__ -std=c++11 -stdlib=libc++ -Wc++11-narrowing -Os -fexceptions -fvisibility=hidden -DNO_LCMS -D__ANSI__ -arch x86_64 -I. -ISource -ISource/Metadata -ISource/FreeImageToolkit -D__ANSI__ -DDISABLE_PERF_MEASUREMENT -I/nix/store/cilmyphkp4qylccn2fg5yilfcm47vmb0-libtiff-4.2.0-dev/include -I/nix/store/b9kr3r3i8fawppc03n053lqf58lz72g2-libjpeg-turbo-2.1.0-dev/include -I/nix/store/wnqimqy7nq7rbpf76jw049ilp2s4xhh0-zlib-1.2.11-dev/include -I/nix/store/bz6y9r2rmmramlavyr5l1r4wsr67swkq-libpng-apng-1.6.37-dev/include/libpng16 -I/nix/store/aff961qxn8w5yn8vh5n1xd5isfsvbwhb-libwebp-1.1.0/include -I/nix/store/b96jp9m78pkkazxpqj1flbi87510w7x9-libraw-0.20.2-dev/include/libraw -I/nix/store/b96jp9m78pkkazxpqj1flbi87510w7x9-libraw-0.20.2-dev/include -I/nix/store/4j2ydnq9fkic9m8x0cqwqlman6qf8f3c-lcms2-2.12-dev/include -I/nix/store/6w8vp1adlq4yqfvfpi44fgla2ms68bsm-openexr-2.5.7-dev/include/OpenEXR -I/nix/store/6w8vp1adlq4yqfvfpi44fgla2ms68bsm-openexr-2.5.7-dev/include -I/nix/store/6w8vp1adlq4yqfvfpi44fgla2ms68bsm-openexr-2.5.7-dev/include/OpenEXR -I/nix/store/3b9cki7cd7i90z7sjmcsrjpmw1v1nl9c-openjpeg-2.4.0-dev/include/openjpeg-2.4 -I/nix/store/6f8f0ny6z1cgw7pxr2a398fjl87y2jk2-jxrlib-1.1/include/libjxr/common -I/nix/store/6f8f0ny6z1cgw7pxr2a398fjl87y2jk2-jxrlib-1.1/include/libjxr/image/x86 -I/nix/store/6f8f0ny6z1cgw7pxr2a398fjl87y2jk2-jxrlib-1.1/include/libjxr/image -I/nix/store/6f8f0ny6z1cgw7pxr2a398fjl87y2jk2-jxrlib-1.1/include/libjxr/glue -I/nix/store/6f8f0ny6z1cgw7pxr2a398fjl87y2jk2-jxrlib-1.1/include/libjxr/test  -mmacosx-version-min=10.11 -c Source/FreeImage/BitmapAccess.cpp -o Source/FreeImage/BitmapAccess.o-x86_64
       > error: unknown target CPU 'armv8.3-a+crypto+sha2+aes+crc+fp16+lse+simd+ras+rdm+rcpc'
       > note: valid target CPU values are: nocona, core2, penryn, bonnell, atom, silvermont, slm, goldmont, goldmont-plus, tremont, nehalem, corei7, westmere, sandybridge, corei7-avx, ivybridge, core-avx-i, haswell, core-avx2, broadwell, skylake, skylake-avx512, skx, cascadelake, cooperlake, cannonlake, icelake-client, icelake-server, tigerlake, knl, knm, k8, athlon64, athlon-fx, opteron, k8-sse3, athlon64-sse3, opteron-sse3, amdfam10, barcelona, btver1, btver2, bdver1, bdver2, bdver3, bdver4, znver1, znver2, x86-64
       > make[1]: *** [Makefile.osx:104: Source/FreeImage/BitmapAccess.o-x86_64] Error 1
       > make[1]: Leaving directory '/private/tmp/nix-build-freeimage-unstable-2020-07-04.drv-0/svn-r1859/FreeImage/trunk'
       > make: *** [Makefile:21: default] Error 2
       For full logs, run 'nix log /nix/store/pbrfippl3qwri32nimx5k0nv28zvhkfw-freeimage-unstable-2020-07-04.drv'.
```


### Steps To Reproduce
Steps to reproduce the behavior:
1. `cd nixpkgs`.  # on an M1 Macbook
2. `nix-build -A freeimage`


### Expected behavior
The build is successful.

### Screenshots
If applicable, add screenshots to help explain your problem.

### Additional context
Add any other context about the problem here.

### Notify maintainers
@uri-canva @viric @L-as 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"aarch64-darwin"`
 - host os: `Darwin 20.6.0, macOS 11.6`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.4`
 - channels(root): `"nixpkgs-21.11pre326916.7053541084b"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/development/libraries/freeimage/default.nix
+++ b/pkgs/development/libraries/freeimage/default.nix
@@ -24,6 +24,9 @@ stdenv.mkDerivation {
       --replace "pkg-config" "$PKG_CONFIG"
     substituteInPlace Makefile.gnu \
       --replace "pkg-config" "$PKG_CONFIG"
+  '' + lib.optionalString (stdenv.isDarwin && stdenv.isAarch64) ''
+    # Upstream Makefile hardcodes i386 and x86_64 architectures only
+    substituteInPlace Makefile.osx --replace "x86_64" "arm64"
   '';
 
   nativeBuildInputs = [

